presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -1032,32 +1046,50 @@ public void testTimestamp(boolean insertWithPresto, ZoneId sessionZone)
+            // is gap friendly.
+            DataType<List<LocalDateTime>> dataType = arrayOfTimestampDataType(precision, insertWithPresto);
+            tests.addRoundTrip(dataType, asList(beforeEpoch));
+            tests.addRoundTrip(dataType, asList(afterEpoch));
+            tests.addRoundTrip(dataType, asList(timeDoubledInJvmZone));
+            tests.addRoundTrip(dataType, asList(timeDoubledInVilnius));
+            tests.addRoundTrip(dataType, asList(epoch));
+            tests.addRoundTrip(dataType, asList(timeGapInJvmZone1));
+            tests.addRoundTrip(dataType, asList(timeGapInJvmZone2));
+            tests.addRoundTrip(dataType, asList(timeGapInVilnius));
+            tests.addRoundTrip(dataType, asList(timeGapInKathmandu));
         }
-        DataTypeTest tests = DataTypeTest.create(true)
-                .addRoundTrip(dataType, asList(beforeEpoch))
-                .addRoundTrip(dataType, asList(afterEpoch))
-                .addRoundTrip(dataType, asList(timeDoubledInJvmZone))
-                .addRoundTrip(dataType, asList(timeDoubledInVilnius))
-                .addRoundTrip(dataType, asList(epoch))
-                .addRoundTrip(dataType, asList(timeGapInJvmZone1))
-                .addRoundTrip(dataType, asList(timeGapInJvmZone2))
-                .addRoundTrip(dataType, asList(timeGapInVilnius))
-                .addRoundTrip(dataType, asList(timeGapInKathmandu));
+
+        // test arbitrary time for all supported precisions
+        tests.addRoundTrip(arrayOfTimestampDataType(1, insertWithPresto), asList(LocalDateTime.of(1970, 1, 1, 1, 1, 1, 100_000_000)));
+        tests.addRoundTrip(arrayOfTimestampDataType(2, insertWithPresto), asList(LocalDateTime.of(1970, 1, 1, 1, 1, 1, 120_000_000)));
+        tests.addRoundTrip(arrayOfTimestampDataType(3, insertWithPresto), asList(LocalDateTime.of(1970, 1, 1, 1, 1, 1, 123_000_000)));
+        tests.addRoundTrip(arrayOfTimestampDataType(4, insertWithPresto), asList(LocalDateTime.of(1970, 1, 1, 1, 1, 1, 123_400_000)));
+        tests.addRoundTrip(arrayOfTimestampDataType(5, insertWithPresto), asList(LocalDateTime.of(1970, 1, 1, 1, 1, 1, 123_450_000)));
+        tests.addRoundTrip(arrayOfTimestampDataType(6, insertWithPresto), asList(LocalDateTime.of(1970, 1, 1, 1, 1, 1, 123_456_000)));
 
         Session session = Session.builder(sessionWithArrayAsArray())
                 .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(sessionZone.getId()))
                 .build();
 
-        tests.execute(getQueryRunner(), session, dataSetup);
+        if (insertWithPresto) {
+            tests.execute(getQueryRunner(), session, prestoCreateAsSelect(sessionWithArrayAsArray(), "test_array_timestamp"));
+        }
+        else {
+            tests.execute(getQueryRunner(), session, postgresCreateAndInsert("tpch.test_array_timestamp"));
+        }
+    }
+
+    private DataType<List<LocalDateTime>> arrayOfTimestampDataType(int precision, boolean insertWithPresto)
```

Comment:
this belong to `Simplify flow in timestamp with timezone mapping tests` commit?

Reply:
no. This is specific to `TIMESTAMP` not `TIMESTAMP w/TZ`. So it belongs to last commit.